File: web/plugins/reverse_string.js
import { WebPlugin } from "../engine/plugin.js";

export class ReverseStringPlugin extends WebPlugin {
    constructor() {
        super("文字を反転");
    }

    async execute(line, context) {
        const match = line.match(/^文字を反転\s+「(.+)」$/);
        if (!match) return;

        context.runtime.write(
            [...match[1]].reverse().join("")
        );
    }
}

File: pc/plugins/reverse_string.py
import re
from core.plugin import Plugin


class ReverseStringPlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(r'文字を反転\s+「(.+)」', line)
        if not match:
            return None

        value = match.group(1)
        return f"print({value!r}[::-1])"